src/components/dialogs/dataSources/COVIDcast.svelte

```diff
@@ -23,38 +26,68 @@
     }
   }
 
-  onMount(() => {
-    fetchCOVIDcastMeta().then((res) => {
-      geoTypes = [...new Set(res.map((d) => d.geo_type))];
-      const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
-      for (const row of res) {
-        const ds = byDataSource.get(row.data_source);
-        if (!ds) {
-          byDataSource.set(row.data_source, {
-            label: row.data_source,
-            value: row.data_source,
-            signals: [row.signal],
-          });
-        } else if (!ds.signals.includes(row.signal)) {
-          ds.signals.push(row.signal);
+  function fetchMetadata() {
+    fetchCOVIDcastMeta((api_key = form_key)).then((res) => {
+      if (res.length !== 0) {
+        invalid_key = false;
+        api_key = form_key; // API key is valid -> use it to fetch data later on
+        geoTypes = [...new Set(res.map((d) => d.geo_type))];
+        const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
+        for (const row of res) {
+          const ds = byDataSource.get(row.data_source);
+          if (!ds) {
+            byDataSource.set(row.data_source, {
+              label: row.data_source,
+              value: row.data_source,
+              signals: [row.signal],
+            });
+          } else if (!ds.signals.includes(row.signal)) {
+            ds.signals.push(row.signal);
+          }
         }
+        byDataSource.forEach((entry) => {
+          entry.signals.sort();
+        });
+        dataSources = [...byDataSource.values()].sort((a, b) => a.value.localeCompare(b.value));
+      } else {
+        invalid_key = api_key != ''; // mark field as invalid, unless it's empty
```

Comment:
An empty key ***is*** a valid key...  Also, what if we inverted the logic of the `if` above and put this shorter case before its alternative?
```suggestion
      } else {
        invalid_key = true
```